Makefile

```diff
@@ -253,8 +329,12 @@ EXPECTED2     :=.krun.expected
 COMMRE = "^\#"
 failing_tests := $(shell grep -v $(COMMRE) tests/failing)
 
+# When tests are run in parallel the TEST_MSG and the subsequent commands
+# may appear in distant lines. It helps keep track of the files being tested
+# though.
```

Comment:
This can be accomplished by just adding the `-O target` option to `make`.

There are other `-O ...` options too.

So this change seems unnecessary to me.

Reply:
Right, `-O` does not produce any change in the output with make version 4.2.1 (that ships with Ubuntu 20.04) but  `--output-sync=recurse` does work with Make version 4.3 (that ships with Ubuntu 21.04). 

So it appears that the PR can be still useful. I added              
```
stage('Simple')                        { steps { sh 'make test-simple -j4 --output-sync=recurse'  } }
```
to Jenkinsfile so if the CI infrastructure is using the newest version of make the output should be properly coalesced.